### ISSUE TITLE
Minor artifacting on Mac OS

### ISSUE THREAD

**Issue by hugeblank** [CCEmuX/CCEmuX#115]
On my MacBook Pro running mac OS Mojave, the corner of the screen (1, 1 on the terminal grid) goes white for about a half a second, and then goes away. 
Another more rare occurrence is when about 1/4 of the first row flashes white, for about the same amount of time. 
I'll look into catching a screenshot of it.

**Comment by Lignum** [comment]
Can we get some details about your system? Java version, graphics card and drivers in particular. Also, does this still happen if you use the JavaFX renderer?

**Comment by hugeblank** [comment]
The JFXPlugin config option was enabled by default. On disabling it the artifacting persisted. I'm not sure if that's what you were referring to when you say 'renderer' so correct me if I just disabled the wrong thing.

Java:
java 11 2018-09-25
Java(TM) SE Runtime Environment 18.9 (build 11+28)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11+28, mixed mode)

Graphics:
Intel Iris Plus Graphics 640

**Comment by dmarcuse** [comment]
The JFX plugin being enabled does not switch the renderer, which renderer is set in the general options? It defaults to the AWT renderer.

**Comment by hugeblank** [comment]
@dmarcuse Ah, AWT was enabled by default, and well... on disabling it I get this fantabulous error:
![image](https://user-images.githubusercontent.com/16393543/46579384-37666d80-c9c5-11e8-91b9-e156ff5028f8.png)


Edit: I really should look into taking windowed screenshots
Edit Edit: I looked into it

**Comment by Lignum** [comment]
Yes, the AWT plugin _provides_ the AWT renderer. The _Renderer_ setting under _CCEmuX Config_ determines which one is actually used. Since that's still set to AWT, which is no longer loaded, CCEmuX no longer has a renderer to use.

To fix this, go into `~/Library/Application Support/ccemux` and delete `ccemux.json`. This will reset it to defaults.

Now, once you've done this, go back into the config dialog, and under _CCEmuX Config_ set _Renderer_ to "JFX", restart the emulator, and tell us if it fixed the issue. Do not disable any of the plugins.

**Comment by hugeblank** [comment]
I cleared the config then tried JFX and JavaFX, both gave me this error (of course the renderer name varied respectively)
![image](https://user-images.githubusercontent.com/16393543/46636031-61c24300-cb0b-11e8-940b-0fe93e8e81d1.png)

**Comment by dmarcuse** [comment]
Java 11 no longer includes JavaFX, which is why you're getting that error. Could you try and get a video recording of the artifacts you're getting with the AWT renderer? I have a few ideas, but I don't have access to a Mac to test with.

**Comment by hugeblank** [comment]
A screen recording is a much smarter idea than what I was going to do, a well timed screenshot.

https://vimeo.com/294078118

Here you go, I catch several instances of the single pixel, but none where a chunk of the row flickers.

Edit: funnily enough the first frame of the video starts when it's in its flickering state

**Comment by Lignum** [comment]
![](https://you.wouldnot.download/a-structural-shoelace.png)
I've screenshotted the issue. It seems to exactly cover the pixel at (1, 1), and is only present in the background layer.

**Comment by Lemmmy** [comment]
Does this happen on a fresh computer without that custom shell on it?

**Comment by hugeblank** [comment]
@Lemmmy Yes, I did realize after making the video I probably should have done it on a fresh one.

**Comment by hugeblank** [comment]
I've found that the blinking stops at the first instance of text it comes across, so in the video, the 'h' is the first thing it comes across so it stops at `(1, 1)`. If I open up a lua terminal, set the cursor position to the middle of the screen, and then clear the screen, the next time it occurs I get half of the screen trying to test if I have epilepsy `((x/2)+1, y/2)` (probably the first and last time I use a math equation to explain what's going on [and I initially did it wrong...]). 

EDIT: video! 
https://vimeo.com/297402149
